bundles/binding/org.openhab.binding.rwesmarthome/src/main/java/org/openhab/binding/rwesmarthome/internal/RWESmarthomeBinding.java

```diff
@@ -0,0 +1,242 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.rwesmarthome.internal;
+
+import java.util.Map;
+
+import org.openhab.binding.rwesmarthome.RWESmarthomeBindingProvider;
+import org.openhab.binding.rwesmarthome.internal.communicator.RWESmarthomeCommunicator;
+import org.apache.commons.lang.StringUtils;
+import org.openhab.core.binding.AbstractActiveBinding;
+import org.openhab.core.events.EventPublisher;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.osgi.framework.BundleContext;
+import org.osgi.service.cm.ConfigurationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+	
+
+/**
+ * Implement this class if you are going create an actively polling service
+ * like querying a Website/Device.
+ * 
+ * @author ollie-dev
+ * @since 1.8.0
+ */
+public class RWESmarthomeBinding extends AbstractActiveBinding<RWESmarthomeBindingProvider> {
+
+	private static final Logger logger = LoggerFactory.getLogger(RWESmarthomeBinding.class);
+	private RWESmarthomeCommunicator communicator = new RWESmarthomeCommunicator();
+	private RWESmarthomeContext context = RWESmarthomeContext.getInstance();
+	
+	/**
+	 * The BundleContext. This is only valid when the bundle is ACTIVE. It is set in the activate()
+	 * method and must not be accessed anymore once the deactivate() method was called or before activate()
+	 * was called.
+	 */
+	private BundleContext bundleContext;
+
+	
+	/** 
+	 * the refresh interval which is used to poll values from the RWESmarthome
+	 * server (optional, defaults to 2000ms)
+	 */
+	private long refreshInterval = 2000;
+	
+	/**
+	 * stores the time when context.setBindingChanged to true
+	 */
+	private long lastBindingChangedTime = 0;
+	
+	
+	public RWESmarthomeBinding() {
+		Runtime.getRuntime().addShutdownHook(new Thread() {
+			@Override
+			public void run() {
+				communicator.stop();
+			}
+		});
```

Comment:
don't use the runtime since your module to hook on is the RWE bundle > rather call stop() in deactivate()